teps/0100-embedded-taskruns-and-runs-status-in-pipelineruns.md

```diff
@@ -171,12 +195,385 @@ In addition to grouping related resources, Tekton Results provides long term sto
-  We will discuss this in the proposal.
-* [Qn #2][qn-1]: What does an example status look like?
-  We will provide examples in the proposal.
+The `PipelineRunStatus` should contain the names and overall status (`ConditionSucceeded`)
+of its `TaskRuns` and `Runs`. Users and tools can find the complete status of `TaskRuns`
+and `Runs` in the cluster. Moreover, they can rely on [Tekton Results](#tekton-results-api)
+and [Owner References and Labels](#owner-references-and-labels) to identify related objects.
+
+### API Changes
+
+#### 1. Add Minimal Embedded Status
+
+Introduce two new structs to store the minimal status of `TaskRuns` and `Runs` in the
+`PipelineRun` status, with the names and overall status only:
+
+```go
+type PipelineRunTaskRunMinimalStatus struct {
+	TaskRunName string `json:"taskRunName,omitempty"`
+	duckv1beta1.Status `json:",inline"`
+}
+
+type PipelineRunRunMinimalStatus struct {
+	RunName string `json:"runName,omitempty"`
+	duckv1beta1.Status `json:",inline"`
+}
+```
+
+Add the new fields to [`PipelineRunStatusFields`](#pipelinerun-status) as follows:
+
+```go
+type PipelineRunStatusFields struct {
+	...
+	TaskRuns map[string]*PipelineRunTaskRunStatus `json:"taskRuns,omitempty"`
+	TaskRunsStatus []PipelineRunTaskRunMinimalStatus `json:"taskRunsStatus,omitempty"`
+	Runs map[string]*PipelineRunRunStatus `json:"runs,omitempty"`
+	RunsStatus []PipelineRunRunMinimalStatus `json:"runsStatus,omitempty"`
+	...
+}
+```
+
+The existing fields providing the complete `TaskRun` and `Runs` are maps with the
+resource names as keys. However, the new fields are sub-objects instead of maps as
+recommended by the [Kubernetes API conventions][subobjects]. 
+
+###### Alternatives
+* [Use Maps](#add-minimal-embedded-status---use-map)
+* [Collapse Fields](#add-minimal-embedded-status---collapse-fields)
+
+#### 2. Deprecate and Remove Full Embedded Status
+
+Deprecate and remove the old fields from `PipelineRunStatusFields` from the Beta API.
+
+```go
+type PipelineRunStatusFields struct {
+	...
+	TaskRunsStatus []PipelineRunTaskRunMinimalStatus `json:"taskRunsStatus,omitempty"`
+	RunsStatus []PipelineRunRunMinimalStatus `json:"runsStatus,omitempty"`
+	...
+}
+```
+
+This is a backwards incompatible change in Beta, therefore the fields will be deprecated
+and removed per out deprecation policy, as described in the [Beta API](#beta-api) section
+below.
+
+### Example
+
+This is an example `PipelineRun` status as provided in the [documentation][example]:
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: PipelineRun
+metadata:
+  ...
+spec:
+    ...
+status:
+    completionTime: "2020-05-04T02:19:14Z"
+    conditions:
+      - lastTransitionTime: "2020-05-04T02:19:14Z"
+        message: "Tasks Completed: 4, Skipped: 0"
+        reason: Succeeded
+        status: "True"
+        type: Succeeded
+    startTime: "2020-05-04T02:00:11Z"
+    taskRuns:
+      triggers-release-nightly-build:
+        pipelineTaskName: build
```

Comment:
what do you think about including the pipelineTask name in the new minimal status fields as well? im guessing it can be inferred from the taskrun name, just wondering if there is any benefit in being explicit

(also could always add this later!)

Reply:
yes, we left it out for that reason (taskrun name = pipelinerun name + pipeline task name)

agreed, we can add it later if needed

made this explicit in the TEP